File: C#/SJN.cs
class SJN_
{
    static void SJN(int[] processes, int[] burstTimes) {
        int n = processes.Length;
        int[,] A = new int[100, 4];
        int total = 0;
        float avgWt, avgTt;

        for (int i = 0; i < n; i++) {
            A[i, 1] = burstTimes[i];
            A[i, 0] = processes[i];
        }

        for (int i = 0; i < n; i++) {
            int index = i;
            for (int j = i + 1; j < n; j++) {
                if (A[j, 1] < A[index, 1])
                {
                    index = j;
                }
            }
            int temp = A[i, 1];
            A[i, 1] = A[index, 1];
            A[index, 1] = temp;

            temp = A[i, 0];
            A[i, 0] = A[index, 0];
            A[index, 0] = temp;
        }

        A[0, 2] = 0;
        for (int i = 1; i < n; i++) {
            A[i, 2] = 0;
            for (int j = 0; j < i; j++) {
                A[i, 2] += A[j, 1];
            }
            total += A[i, 2];
        }
        avgWt = (float)total / n;

        total = 0;
        Console.WriteLine("P\tBurst Time\tWaiting Time\tTurnaround Time");
        for (int i = 0; i < n; i++) {
            A[i, 3] = A[i, 1] + A[i, 2];
            total += A[i, 3];
            Console.WriteLine("P" + A[i, 0] + "\t" + A[i, 1] + "\t" + A[i, 2] + "\t" + A[i, 3]);
        }
        avgTt = (float)total / n;

        Console.WriteLine("Average Waiting Time= " + avgWt);
        Console.WriteLine("Average Turnaround Time= " + avgTt);
    }

    static void Main(string[] args) {
        int[] processes = [1, 2, 3];
        int[] burstTimes = [10, 5, 8];
        SJN(processes, burstTimes);
    }
}

File: Python/SJN.py
def SJN(processes, burstTime):
    n = len(processes)
    A = [[0 for j in range(4)] for i in range(100)]
    total, avgWt, avgTat = 0, 0, 0

    for i in range(n):
        A[i][1] = burstTime[i]
        A[i][0] = processes[i]

    for i in range(n):
        index = i
        for j in range(i + 1, n):
            if A[j][1] < A[index][1]:
                index = j

        A[i][1], A[index][1] = A[index][1], A[i][1]
        A[i][0], A[index][0] = A[index][0], A[i][0]

    A[0][2] = 0
    for i in range(1, n):
        A[i][2] = sum(A[j][1] for j in range(i))
        total += A[i][2]
    avgWt = total / n

    total = 0
    print("P\tBurst Time\tWaiting Time\tTurnaround Time")
    for i in range(n):
        A[i][3] = A[i][1] + A[i][2]
        total += A[i][3]
        print(f"P{A[i][0]}\t{A[i][1]}\t{A[i][2]}\t{A[i][3]}")
    avgTat = total / n

    print(f"Average Waiting Time= {avgWt}")
    print(f"Average Turnaround Time= {avgTat}")


if __name__ == "__main__":
    processes = [1, 2, 3]
    burstTime = [10, 5, 8]
    SJN(processes, burstTime)
